### DISCUSSION TITLE
Fail to start RELEASE.2023-09-30T07-02-29Z with special characters in access_key and secret_key

### DISCUSSION THREAD

**Discussion by bengtfredh** [minio/minio#18147] [Q&A] [answered]
## NOTE
RELEASE.2023-09-30T07-02-29Z fail to start with special character in access_key and secret_key

## Expected Behavior
Until this version minio can run with special character in access_key and secret_key

## Current Behavior
minio give error:
```
ERROR Unable to validate credentials inherited from the shell environment: Invalid credentials
      > Please provide correct credentials
      HINT:
        Access key length should be at least 3, and secret key length at least 8 characters
```

## Possible Solution
Run with no special character in access_key and secret_key

## Steps to Reproduce (for bugs)

1. Provide access_key and secret_key with special character 
2. Start minio

## Context
minio is not starting
replaced access_key and secret_key with no special character 

## Regression
Is this issue a regression? Yes
Looks like this issue has happen before:
https://github.com/minio/minio/pull/10223

## Your Environment

* Version used (`minio --version`): RELEASE.2023-09-30T07-02-29Z
* Server setup and configuration: podman
* Operating System and version (`uname -a`): `Linux 6.5.5-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Sun Sep 24 15:52:44 UTC 2023 x86_64 GNU/Linux`

**Comment by harshavardhana** [comment]
@bengtfredh what is the special character?

**Reply by bengtfredh** [reply]
access_key : U4B6Zi!b75DXSPm%avZb
secret_key: Q4#Q6y8G%%Px#npP4dudUobU#NBcGB7RMKV4ajYb

**Comment by harshavardhana** [comment, marked as answer]
Runs perfectly fine looks like you have some local issue on your end you need to fix

```sh
#!/bin/bash

export MINIO_ROOT_USER="U4B6Zi!b75DXSPm%avZb"
export MINIO_ROOT_PASSWORD="Q4#Q6y8G%%Px#npP4dudUobU#NBcGB7RMKV4ajYb"

minio server /tmp/fresh
```

```
~/run-special.sh 
Formatting 1st pool, 1 set(s), 1 drives per set.
WARNING: Host local has more than 0 drives of set. A host failure will result in data becoming unavailable.
MinIO Object Storage Server
Copyright: 2015-2023 MinIO, Inc.
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Version: DEVELOPMENT.2023-09-30T10-13-56Z (go1.21.1 linux/amd64)

Status:         1 Online, 0 Offline. 
S3-API: http://10.0.0.185:9000  http://172.17.0.1:9000  http://172.19.0.1:9000  http://172.23.0.1:9000  http://127.0.0.1:9000               
RootUser: U4B6Zi!b75DXSPm%!!(MISSING)a(MISSING)vZb 
RootPass: Q4#Q6y8G%!P(MISSING)x#npP4dudUobU#NBcGB7RMKV4ajYb 

Console: http://10.0.0.185:45541 http://172.17.0.1:45541 http://172.19.0.1:45541 http://172.23.0.1:45541 http://127.0.0.1:45541        
RootUser: U4B6Zi!b75DXSPm%!!(MISSING)a(MISSING)vZb 
RootPass: Q4#Q6y8G%!P(MISSING)x#npP4dudUobU#NBcGB7RMKV4ajYb 

Command-line: https://min.io/docs/minio/linux/reference/minio-mc.html#quickstart
   $ mc alias set myminio http://10.0.0.185:9000 U4B6Zi!b75DXSPm%!a(MISSING)vZb Q4#Q6y8G%Px#npP4dudUobU#NBcGB7RMKV4ajYb

Documentation: https://min.io/docs/minio/linux/index.html
Warning: The standard parity is set to 0. This can lead to data loss.
```

**Reply by klauspost** [reply]
> `Q4#Q6y8G%!P(MISSING)x#npP4dudUobU#NBcGB7RMKV4ajYb `

The (MISSING) part seems to be a problem.

**Reply by klauspost** [reply]
(it is just a printing issue, though)
